springBoot最終課題の提出

Azuma-Chihiro opened the pull request:
#### がんばりました。

Azuma-Chihiro committed:
springBoot最終課題の提出

Azuma-Chihiro committed:
最終課題の再提出

Azuma-Chihiro commented:
修正、再提出

Azuma-Chihiro committed:
最終課題の修正。

Azuma-Chihiro commented:
##### springSecurityの導入
ログイン時に権限がないUserは入れない設定にし、ログアウト完了後も権限がないUserはログインページにとぶように設定。

##### 編集画面のValidation
空欄で保存すると入力フォームに返ってくる。

uchiyama4929 commented on practice/src/main/java/com/example/demo/config/WebSecurityConfig.java:
```diff
@@ -0,0 +1,74 @@
+				//.successForwardUrl("/admin/contacts")
+				.permitAll()
+				)
+		.csrf(csrf -> csrf.disable())//無効か 書かなくても動く処理、制限がかかっていても通るようにする
```
csrfチェックを無効化しているので有効化に変更して動作するようにしてください。

uchiyama4929 commented on practice/src/main/java/com/example/demo/config/WebSecurityConfig.java:
```diff
@@ -0,0 +1,74 @@
+				.permitAll()
+				.requestMatchers("/contacts/**").permitAll()
+				//.authenticated()
+				//.anyRequest()
```
不要なコメントはなるべく消すようにしてください。

uchiyama4929 commented on practice/src/main/java/com/example/demo/controller/AdminController.java:
```diff
@@ -0,0 +1,118 @@
+	/*contactsDBから一覧を表示
+	*findAll();
+	*contact情報一覧画面 return 情報一覧;
+	*/
```
JavaDocの構文は

	/**
	 * contactsDBから一覧を表示
	 * findAll();
	 * contact情報一覧画面 return 情報一覧;
	 */

といった感じになります。
/** で開始するとエディタ上での表示色も変わるかと思うので確認してみてください。

uchiyama4929 reviewed (changes requested):
一旦コメントにて修正箇所を残しておきます。
こちら側でブラウザ上での動作確認ができていないので追加で指摘事項があるかもしれません。

uchiyama4929 commented:
ログアウトボタンで戻った後
http://localhost:8080/admin/contacts/1/edit などの画面に飛ぶことが出来るので修正をお願いします！

uchiyama4929 commented:
お問い合わせ編集画面でバリデーションに失敗した場合のエラーメッセージを表示してください。

Azuma-Chihiro committed:
csrfの処理はデフォルトで有効のため記述を消去
不要なコメントを消去

Azuma-Chihiro commented:
ご確認よろしくお願い致します。

Azuma-Chihiro committed:
追記、再コミット
編集画面のエラーメッセージの記載漏れ
contactServiceでContactをcontactFormに変換するgiveContactMethodを作り、contorllerでcontactEditをcontactFormに入れedit.htmlのth:objectにcontactFormを入れエラー文を実装

uchiyama4929 commented:
> ログアウトボタンで戻った後
> http://localhost:8080/admin/contacts/1/edit などの画面に飛ぶことが出来るので修正をお願いします！

こちらの修正をお願いします。
また、 http://localhost:8080/contact  の画面も403エラーになることがありますが、こちらの画面は管理者ログインとは無関係で常時表示できる状態にしてください！

Azuma-Chihiro committed:
springSecurityの修正

Azuma-Chihiro commented:
springSecurityを修正し、/contactには誰でも入れるように修正
管理者フォームはログイン認証で表示に変更。

##### ご確認よろしくお願い致します。
